src/app/(dashboard)/dashboard/payment/page.tsx

```diff
@@ -1,39 +1,76 @@
 'use client';
-import { useEffect } from 'react';
-import { ReadonlyURLSearchParams, useRouter, useSearchParams } from 'next/navigation';
-import { toast } from 'react-toastify';
 
-interface SearchParams extends ReadonlyURLSearchParams {
-  type?: string;
-  error?: string;
-}
+import { Button } from '@/components/UI';
+import { PaymentElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { StripePaymentElementOptions } from '@stripe/stripe-js';
+import { useState } from 'react';
+import styles from './style.module.scss';
 
 const Payment = () => {
-  const router = useRouter();
-  const query: SearchParams = useSearchParams();
-
-  useEffect(() => {
-    if (query.type === 'success') {
-      toast.success('Paiement effectué avec succès');
-    } else if (query.type === 'error') {
-      switch (query.error) {
-        case 'CART_NOT_FOUND':
-          toast.error('Panier introuvable');
-          break;
-        case 'TRANSACTION_ERROR':
-          toast.error('La transaction a échoué');
-          break;
-        case 'NO_PAYLOAD':
-          toast.error('Requête erronée');
-          break;
-        default:
-          break;
-      }
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [message, setMessage] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!stripe || !elements) {
+      // Stripe.js hasn't yet loaded.
+      // Make sure to disable form submission until Stripe.js has loaded.
+      return;
     }
-    router.push('/dashboard');
-  }, []);
 
-  return false;
+    setIsLoading(true);
+
+    const { error } = await stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        // Make sure to change this to your payment completion page
+        return_url: process.env.NEXT_PUBLIC_URL + '/dashboard/payment/callback',
+      },
+    });
+
+    // This point will only be reached if there is an immediate error when
+    // confirming the payment. Otherwise, your customer will be redirected to
+    // your `return_url`. For some payment methods like iDEAL, your customer will
```

Comment:
et same ici, ça serait bien d'adapter le comment à ce qu'on a la (genre pas "your customer" mais "the customer", pas "your return_url" mais "/dashboard/payment/callback" ou "l'url de callback"